### ISSUE TITLE
Test Cases for Redis or design improvement

### ISSUE THREAD

**Issue by kumaranshu72** [kumaranshu72/spotter#11]
Can some help me with writing test cases for Redis or even improve the way I am writing the test cases. Also it would be great if someone can help with improving the LLD.

**Comment by mythio** [comment]
@kumaranshu72 let me give it a shot! :smirk_cat:

**Comment by kumaranshu72** [comment]
@mythio go ahead

**Comment by mythio** [comment]
Okay let me tell you what I did to run this project. I have `docker` and `docker-compose` installed. I cloned your repo, and then
```bash
npm config set save=true
npm config set save-exact=true
docker-compose -f docker-compose.yml up --build
```
But this won't run! It doesn't connect to redis or mongo. When I press Ctrl+C, and then again do docker-compose, it just throws error saying, the `./data/db` doesn't have permissions to be used.

What am I missing here?

**Comment by kumaranshu72** [comment]
Give 777 permission to data directory.

**Comment by kumaranshu72** [comment]
Also create a .env file similar to .env_samples

**Comment by mythio** [comment]
There is no file as `.env_samples` in the project.

**Comment by kumaranshu72** [comment]
The path for env sample is `config/env/environment-sample.ts`. Change the environment-sample.ts name to the environment.ts

**Comment by mythio** [comment]
Yes I have done that too. Are you sure this project runs as it is in the master branch? Or are there errors put purposely? I checked docker-compose.yml and it had wrong ports exposed. Can you explain why is that so?

**Comment by kumaranshu72** [comment]
I have updated the `environment-sample.ts`, please update your environment.ts accordingly

**Comment by ghost** [comment]
Perfect! Works like a charm after this.

**Comment by kumaranshu72** [comment]
@mythio that was the miss at our end. I will update readme and contribution.md with the above change. Sorry for the inconveniences.
